acceptance/tests/utf8/utf8-in-catalog.rb

```diff
@@ -0,0 +1,132 @@
+test_name 'utf-8 characters in cached catalog' do
+  # utf8chars = "€‰ㄘ万竹ÜÖ"
+  utf8chars = "\u20ac\u2030\u3118\u4e07\u7af9\u00dc\u00d6"
+  file_content = "This is the file content. file #{utf8chars}"
+  tmpdir = master.tmpdir("code")
+  on(master, "rm -rf #{tmpdir}")
+  env_dir = "#{tmpdir}/environments"
+  agents.each do |agent|
+    agent_file = agent.tmpfile("file" + utf8chars) 
+
+    step "Apply manifest" do
+      on(
+        agent,
+        "rm -rf #{agent_file}",
+        :environment => {:LANG => "en_US.UTF-8"}
+      )
+    
+      master_manifest =
+<<PP
+    
+File {
+  ensure => directory,
+  mode => "0755",
+}
+
+file {
+  '#{tmpdir}/':;
+  '#{tmpdir}/environments':;
+  '#{tmpdir}/environments/production':;
+  '#{tmpdir}/environments/production/manifests':;
+}
+
+file { '#{env_dir}/production/manifests/site.pp' :
+  ensure => file,
+  mode => '0644',
+  content => '
+file { "#{agent_file}" :
+  ensure => file,
+  mode => "0644",
+  content => "#{file_content}
+",
+}
+',
+}
+
+PP
+        
+      apply_manifest_on(
+        master,
+        master_manifest,
+        {
+          :acceptable_exit_codes => [0, 2],
+          :catch_failures => true,
+          :environment => {:LANG => "en_US.UTF-8"}
+        }
+      )
+    end
+
+    master_opts = {
+      'main' => {
+        'environmentpath' => "#{env_dir}",
+      },
+      'agent' => {
+        'use_cached_catalog' => 'true'
+      }
+    }
+    
+    with_puppet_running_on(master, master_opts, tmpdir) do 
+      step "puppet agent -t" do
+        on(
+          agent,
+          puppet("agent -t", "--server #{master.hostname}"),
+          {
+            :acceptable_exit_codes => [0, 2],
+            :environment => {:LANG => "en_US.UTF-8"}
+          }
+        )
+      end
+    
+      step "verify cached catalog" do
+        result = on(
+          agent,
+          puppet("config print vardir"),
```

Comment:
Rather than having to do an extra ssh call out to the agent to gets its `vardir`, you can just tell the agent what its vardir is with `--vardir` in the preceding `puppet("agent -t"...` call. Saves us both the extra time of the call and yet another possible ssh/network failure. 

An example of this being done is here:
https://github.com/puppetlabs/puppet/blob/master/acceptance/tests/direct_puppet/catalog_uuid_correlates_catalogs_with_reports.rb#L8-L30